### PR TITLE
refactored Youtube page.svelte code following TS best practices

### PR DESCRIPTION
**Pull request by imolina212**
## Describe your changes

- Cleaned up typing and console.logs
- Added Channel interface
- Renamed functions and variable names to be more descriptive 
- Changed variable declarations from let to const

## Related issue number/link
N/A

---
Thanks for contributing! 🥳🚀🌳

### CODE REVIEW

File: src/routes/youtube/+page.svelte [hunk 1]
```diff
@@ -4,73 +4,75 @@
...
+  const channelData = data.payload.channelData;
   let displayedVideos = videoData;
 
   let rerender: boolean = false;
+  
+  interface Channel {
+    channelId: string;
+    channelSubCount: number;
+  }
+  interface channelFilterItem {
```
Comment (williamtaggart97) [inline src/routes/youtube/+page.svelte:17]:
```suggestion
  interface ChannelFilterItem { 
```

File: src/routes/youtube/+page.svelte [hunk 2]
```diff
@@ -4,73 +4,75 @@
...
-    console.log(displayedVideos);
 
     // Force svelte re-render
     rerender = !rerender;
   }
 
-  function semanticNumber(number: number) {
-    // if number between 1 and 99
-    if (number < 1000) {
+  function formatChannelSubCount(num: number) {
```
Comment (williamtaggart97) [inline src/routes/youtube/+page.svelte:70]:
```suggestion
  function formatChannelSubCount(subCount: number) {
```

File: src/routes/youtube/+page.svelte [hunk 3]
```diff
@@ -4,73 +4,75 @@
...
-  let filterObject: channelFilterItem[] = [];
-  for (let channelInfo of channelData) {
-    filterObject.push({ channelId: channelInfo.channelId, active: true });
+  const channelArr: channelFilterItem[] = [];
+  for (const channelInfo of channelData) {
+    channelArr.push({ channelId: channelInfo.channelId, active: true });
   }
 
-  function filterResources(event) {
+  function filterResources() {
```
Comment (williamtaggart97) [inline src/routes/youtube/+page.svelte:49]:
```suggestion
  function filterVideos(videoData: Video[], channelArr: ChannelFilterItem[]): Video[] {
```

Comment (williamtaggart97) [reply src/routes/youtube/+page.svelte:49]: Pure functions are better :)

File: src/routes/youtube/+page.svelte [hunk 4]
```diff
@@ -4,81 +4,82 @@
...
     const ClimateTownChannelId = "UCuVLG9pThvBABcYCm7pkNkA";
+    
     if (a.channelId === ClimateTownChannelId) {
       return -1;
     }
     if (b.channelId === ClimateTownChannelId) {
       return 1;
     }
 
-    // Normal sort
```
Comment (VeckoTheGecko) [inline src/routes/youtube/+page.svelte:23]: Please keep this comment (even if its just for me and my Typescript noobiness :) )

File: src/routes/youtube/+page.svelte [hunk 5]
```diff
@@ -4,81 +4,82 @@
...
+    channelSubCount: number;
+  };
+  interface ChannelFilterItem { 
+    channelId: string;
+    active: boolean;
+  };
+  interface Video {
+    channelId: string;
+    [key: string]: any;
+  };
```
Comment (VeckoTheGecko) [inline src/routes/youtube/+page.svelte:24]: Could we move this into the `interfaces.ts` file?

File: src/routes/youtube/+page.svelte [hunk 6]
```diff
@@ -4,81 +4,82 @@
...
-    // if number between 1 and 99
-    if (number < 1000) {
+  function formatChannelSubCount(subCount: number) {
+    if (subCount < 1000) {
       return "<1k";
     }
-    if (number >= 1000) {
-      return `${Math.floor(number / 1000)}k`;
+    if (subCount >= 1000) {
+      return `${Math.floor(subCount / 1000)}k`;
```
[the remark this answers is not in the document]
Comment (VeckoTheGecko) [reply src/routes/youtube/+page.svelte]: Done in c031cb0 :)

[the remark this answers is not in the document]
Comment (imolina212) [reply src/routes/youtube/+page.svelte]: Makes sense, will do 👍

File: src/routes/youtube/+page.svelte (hunk 5 above)
Comment (imolina212) [reply src/routes/youtube/+page.svelte:24]: That's a great idea. I'll take care of that 👍

**Pull request closed without merging**